underfs/b2/src/main/java/alluxio/underfs/b2/B2InputStream.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.b2;
+
+import com.backblaze.b2.client.B2StorageClient;
+import com.backblaze.b2.client.contentHandlers.B2ContentMemoryWriter;
+import com.backblaze.b2.client.exceptions.B2Exception;
+import com.backblaze.b2.client.structures.B2DownloadByNameRequest;
+import com.backblaze.b2.util.B2ByteRange;
+import com.google.common.base.Throwables;
+import com.google.protobuf.ServiceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * A stream for reading a file from B2.
+ */
+@NotThreadSafe public class B2InputStream extends InputStream {
+  private static final Logger LOG = LoggerFactory.getLogger(B2InputStream.class);
+  private final String mBucketName;
+  private final String mKey;
+  private final B2StorageClient mB2StorageClient;
+  private long mOffset;
+  private long mLength;
+  private BufferedInputStream mBufferedInputStream;
+
+  /**
+   * Constructor for an input stream of an object in B2 using the B2-sdk implementation to read
+   * data.
+   *
+   * @param bucketName      the bucket the object resides in
+   * @param key             the path of the object to read
+   * @param b2StorageClient the b2 client to use for operations
+   * @param length          the length of the stream
```

Comment:
If this is in bytes, can you specify that to this javadoc comment? (Also applies to other javadoc fields in this file)

Reply:
Fixed